### DISCUSSION TITLE
Can't save new password: "Error from OpenKeyChain: missing key for encryption"

### DISCUSSION THREAD

**Discussion by np2048** [zeapo/Android-Password-Store#1304] [Q&A]
The application fails to save a new password. It displays an error:
Error from OpenKeyChain: missing key for encryption

I've imported public.key file in OpenKeyChain applications and the private key. 

Password Store app version: 1.13.2

**Comment by msfjarvis** [comment]
You need to import the private part of your GPG key in OpenKeychain.

**Reply by np2048** [reply]
> You need to import the private part of your GPG key in OpenKeychain.

Private key was imported too. I am able to decrypt passwords from the storage but not to create new passwords.

**Reply by msfjarvis** [reply]
It might be this issue then: https://github.com/open-keychain/open-keychain/issues/1781. If revoking access does not work, try clearing OpenKeychain's data and re-importing your keys.

**Reply by np2048** [reply]
> It might be this issue then: [open-keychain/open-keychain#1781](https://github.com/open-keychain/open-keychain/issues/1781). If revoking access does not work, try clearing OpenKeychain's data and re-importing your keys.

Unfortunately it doesn't help. I've deleted keys and even all OpenKeychain application data and reimported the keys, revoked access for the Passwod Store application and grant it again but it still the same. Done that a couple of times. In the topic you've send the link to it is said that the issue with Password Store occurs with a lot of users and the solution described there doesn't help:

> And the same thing happens now years later but with PassworStore. I already revoked access and tried again.

It is also said there:

> These inconsistencies can have multiple reasons [...] This whole issue goes away when apps no longer use the deprecated, account-based api.

**Reply by msfjarvis** [reply]
> It is also said there:
> 
> > These inconsistencies can have multiple reasons [...] This whole issue goes away when apps no longer use the deprecated, account-based api.

We don't use the deprecated API. In the OP, you will notice the log snippet that contains this line

```
You are using a deprecated service which may lead to truncated data on return, please use IOpenPgpService2
```

Searching for `IOpenPgpService2`, you'll find it is [used throughout our codebase](https://github.com/android-password-store/Android-Password-Store/search?q=IOpenPgpService2).

That being said, there isn't anything that we can do about this. It's either a problem with your key or an OpenKeychain bug.

**Reply by tennox** [reply]
After debugging & solving https://github.com/open-keychain/open-keychain/issues/2588#issuecomment-1092920924 I thought you might have a similar situation - do you have multiple keys in `.gpg-id` ?

**Comment by zachdoestein** [comment]
I had the same issue, but a work around was to create a new folder within the app and check off the "Set GPG key for directory" box then move the old or create the new passwords within that folder. This creates a new .gpg-id within the new folder. (Importing all the public keys didn't work for me as suggested by [this](https://github.com/open-keychain/open-keychain/issues/2588#issuecomment-1092920924))

**Reply by ulfi** [reply]
> I had the same issue, but a work around was to create a new folder within the app and check off the "Set GPG key for directory" box then move the old or create the new passwords within that folder. This creates a new .gpg-id within the new folder.

At my side the same. I copied the old folders and after creating a new one, I was ask to select the keys. Afterwards I copied all old GPG files back. This works for me as well :smile:
